draft-packages/modal/KaizenDraft/Modal/Primitives/ModalFooter.tsx

```diff
@@ -25,18 +20,15 @@ const ModalFooter: ModalFooter = props => {
   return (
     <GenericModalSection unpadded={unpadded}>
       <div className={styles.actions}>
-        {actions.map((a, index) => (
-          <div className={styles.actionButton} key={a.label}>
+        {actions.map((action, index) => (
+          <div className={styles.actionButton} key={action.label}>
             <Button
               type="button"
-              icon={a.icon}
-              label={a.label}
-              onClick={a.action}
               primary={index === 0 && appearance === "primary"}
               destructive={index === 0 && appearance === "destructive"}
               secondary={index > 0}
-              disabled={a.disabled}
               data-automation-id={`${automationId}-action-${index}`}
+              {...action}
```

Comment:
Props on components have precedence, meaning this `action` on line 31 will override any of the duplicate props defined above it. So if `appearance` or `data-automation-id` is included in `action` (possible, since it's extending ButtonProps) those props will be overridden. 

It's best to bump `{...action}` up to line 26.

Reply:
(unless of course it's intended to be overridden)

Reply:
Yeah I was actually thinking to allow these to be overridden.

Reply:
Cool - wasn't sure if it was intentional. I wonder if we _should_ be enforcing order of buttons on modals. it's not the best UI to have cancel buttons in different positions on different modals. 🤔 One for the future, not now.